### PR TITLE
Update conditional logic for editor_styles

### PR DESCRIPTION
**Pull request by mikachan**
Trac ticket: https://core.trac.wordpress.org/ticket/57561

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by hellofromtonya** [comment]
Tests for failing for

```
Tests_Dependencies_Styles::test_block_styles_for_editing_without_theme_support
Failed asserting that false is true.
```

for the last assertion in this test:
```php
	public function test_block_styles_for_editing_without_theme_support() {
		// Confirm we are without theme support by default.
		$this->assertFalse( current_theme_supports( 'wp-block-styles' ) );

		wp_default_styles( $GLOBALS['wp_styles'] );

		$this->assertFalse( wp_style_is( 'wp-block-library-theme' ) );
		wp_enqueue_style( 'wp-edit-blocks' );
		$this->assertTrue( wp_style_is( 'wp-block-library-theme' ) );
	}
```

Looking at the code in this PR, the `'wp-block-library-theme'` style is no longer enqueued if the theme doesn't support `'wp-block-styles'`. This means the last assertion in this test should expect the state the state not to change, i.e. `assertFalse()`. 

In reading the intent of the [original Gutenberg PR](https://github.com/WordPress/gutenberg/pull/45351):
>This PR is an attempt to never allow the `wp-block-library-theme` CSS to be loaded under the conditions we introduced in https://github.com/WordPress/gutenberg/pull/44640.

this does appear to be the intent, i.e. if the theme doesn't support `'wp-block-styles'`, then `wp_style_is( 'wp-block-library-theme' )` should be `false`. 

I'll update the unit test accordingly.

**Comment by mikachan** [comment]
Thank you for updating that test, @hellofromtonya 🙇‍♂️

> if the theme doesn't support 'wp-block-styles', then wp_style_is( 'wp-block-library-theme' ) should be false.

This is correct 👍

**Review by hellofromtonya — APPROVED** [review]
Tested for both TT2 and TT3 https://core.trac.wordpress.org/ticket/57561#comment:13

* Fixes the issue for TT3, a theme that does not add `'wp-block-styles'` theme support ✅ 
* Does not impact TT2, a theme that does add `'wp-block-styles'` theme support ✅ 

This PR is ready for commit 👍

File: src/wp-includes/script-loader.php [hunk 1]
```diff
@@ -1624,8 +1624,11 @@ function wp_default_styles( $styles ) {
 		$wp_edit_blocks_dependencies[] = 'wp-editor-classic-layout-styles';
 	}
 
-	if ( ! is_array( $editor_styles ) || count( $editor_styles ) === 0 ) {
-		// Include opinionated block styles if no $editor_styles are declared, so the editor never appears broken.
+	if (
+		current_theme_supports( 'wp-block-styles' ) &&
+		( ! is_array( $editor_styles ) || count( $editor_styles ) === 0 )
+	) {
+		// Include opinionated block styles if the theme supports block styles and no $editor_styles are declared, so the editor never appears broken.
```
Comment (hellofromtonya) [inline src/wp-includes/script-loader.php]:
```suggestion
		/*
		 * Include opinionated block styles if the theme supports block styles and
		 * no $editor_styles are declared, so the editor never appears broken.
		 */
```

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/55368.

**Pull request closed without merging**